src/editor.c

```diff
@@ -1906,9 +2002,27 @@ static gchar *find_calltip(const gchar *word, GeanyFiletype *ft)
 
 	g_return_val_if_fail(ft && word && *word, NULL);
 
-	/* use all types in case language uses wrong tag type e.g. python "members" instead of "methods" */
-	tags = tm_workspace_find(word, NULL, tm_tag_max_t, NULL, ft->lang);
-	if (tags->len == 0)
+	tags = get_scoped_tags(editor, pos - strlen(word));
+	if (tags)
+	{
+		GPtrArray *filtered = g_ptr_array_new();
+
+		foreach_ptr_array(tag, i, tags)
+		{
+			if (strcmp(tag->name, word) == 0)
+				g_ptr_array_add(filtered, tag);
```

Comment:
@kugel- This is a good example why I suggested the new query interface operates on tag arrays instead - especially with scope completion you have to work with tag arrays which are results of a much more complicated query than what your query interface can handle and further filtering isn't possible using your API.